### ISSUE TITLE
GridView multiple rowOptions

### ISSUE THREAD

**Issue by alex-code** [yiisoft/yii2#11544]
Would something like this be useful in the core `GridView` widget?
This allows columns to add row options so bootstrap contextual classes for tables can be used.

``` php
<?php

namespace backend\grid;

use yii\helpers\Html;

class GridViewEx extends \yii\grid\GridView
{
    private $_rowOptions = [];
    public function getRowOptions()
    {
        return $this->_rowOptions;
    }

    public function setRowOptions($options)
    {
        $this->_rowOptions[] = $options;
    }

    public function renderTableRow($model, $key, $index)
    {
        $cells = [];
        foreach ($this->columns as $column) {
            $cells[] = $column->renderDataCell($model, $key, $index);
        }
        $options = [];
        foreach ($this->_rowOptions as $rowOptions) {
            if ($rowOptions instanceof \Closure) {
                $options = array_merge_recursive($options, call_user_func($rowOptions, $model, $key, $index, $this));
            } else {
                $options = array_merge_recursive($options, $rowOptions);
            }
        }
        $options['data-key'] = is_array($key) ? json_encode($key) : (string) $key;

        return Html::tag('tr', implode('', $cells), $options);
    }
}
```

a column can then specify row options

``` php
<?php

namespace common\columns;

class StatusColumn extends \yii\grid\DataColumn
{
    public function init()
    {
        parent::init();
        $this->grid->rowOptions = function ($model, $key, $index, $grid) {
            if (STATUS_DISABLED === $model->status) {
                return ['class' => 'active'];
            }
            return [];
        };
    }
}
```

**Comment by petrabarus** [comment]
What's the difference with `rowOptions`?

http://www.yiiframework.com/doc-2.0/yii-grid-gridview.html#$rowOptions-detail

**Comment by Alex-Code** [comment]
If you have any custom columns they can add to the `rowOptions` so you can have contextual colours etc.

**Comment by samdark** [comment]
Won't be implemented in 2.0.
